PegSolitaire.cpp

```diff
@@ -0,0 +1,207 @@
+﻿
+#include "PegSolitaire.hpp"
+
+const std::array<std::array<char, 9>, 9> ps::get_english_board() {
+	std::fstream file("../PegSolitaireBoard.txt");
+	std::array<std::array<char, 9>, 9> board = {};
+	if (file.is_open()) {
+		std::cout << "Reading file" << std::endl;
+		std::string line;
+		int row = 0;
+		while (std::getline(file, line)) {
+			auto s = line.size();
+			for (int col = 0; col < line.size(); col++) {
+				auto charValue = line[col];
+				if (charValue == 'E') {
+					return board;
+				}
+				else if (charValue == '1') {
+					board[row][col] = '1';
+				}
+				else if (charValue == '0') {
+					board[row][col] = '0';
+				}
+				else {
+					board[row][col] = ' ';
+				}
+			}
+			row++;
+		}
+	}
+	else {
+		std::cerr << "Error opening file" << std::endl;
+	}
+	return board;
+}
+void ps::print(const std::array<std::array<char, 9>, 9>* board) {
+	for (auto& line : *board) {
+		for (auto& chr : line) {
+			std::cout << chr;
+		}
+		std::cout << std::endl;
+	}
+}
+void updatePath(std::vector<std::array<char, 5>>* path, const char lastDirection,
+	const char currentRow, const char currentCol, const char rowOffset, const char colOffset) {
+	std::array<char, 5> temp = {};
+	temp[0] = lastDirection;
+	temp[1] = currentRow;
+	temp[2] = currentCol;
+	temp[3] = currentRow + rowOffset;
+	temp[4] = currentCol + colOffset;
+	path->push_back(temp);
+}
+static bool check_X_away(const std::array<std::array<char, 9>, 9>* board, const char currentRow, const char currentCol,
+	const char rowOffset, const char colOffset, const char checkValue) {
+	if ((*board)[currentRow + rowOffset][currentCol + colOffset] == checkValue) {
+		return true;
+	}
+	return false;
+}
+static char remainingPieces(const std::array<std::array<char, 9>, 9>* board) {
+	char remainingPieces = 0;
+	for (auto& line : *board) {
+		for (auto& chr : line) {
+			if (chr == '1') {
+				++remainingPieces;
+				if (remainingPieces > 1) {
+					return remainingPieces;			// Return even if you don't count them all
+				}
+			}
+		}
+	}
+	return remainingPieces;

```

Comment:
... can be then simplified
```suggestion
	for (auto& line : *board) {
		for (auto& chr : line) {
			if (chr == '1') {
                            return true;
			}
		}
	}
	return false;
```

Reply:
How will I check if board is completed if there will always be 1 last peg on the board?